doctests fail with GHC 9.2.1 and doctest 0.18.2

```
Test suite doctests: RUNNING...

Data/ByteString/Builder/HTTP/Chunked.hs:117:47: error:
    • Found type wildcard ‘_x’ standing for ‘a1’
      Where: ‘a1’ is a rigid type variable bound by
               the inferred type of
                 go :: (BufferRange -> IO (BuildSignal a1)) -> BuildStep a
               at Data/ByteString/Builder/HTTP/Chunked.hs:(118,9)-(177,64)
      To use the inferred type, enable PartialTypeSignatures
    • In the type signature:
        go :: (BufferRange -> IO (BuildSignal _x)) -> BuildStep a
      In an equation for ‘transferEncodingStep’:
          transferEncodingStep k
            = go (B.runBuilder innerBuilder)
            where
                go :: (BufferRange -> IO (BuildSignal _x)) -> BuildStep a
                go innerStep (BufferRange op ope)
                  | outRemaining < minimalBufferSize
                  = pure $ B.bufferFull minimalBufferSize op (go innerStep)
                  | otherwise
                  = B.fillWithBuildStep innerStep doneH fullH insertChunkH brInner
                  where
                      outRemaining = ope `F.minusPtr` op
                      maxChunkSizeLength = word32HexLength $ fromIntegral outRemaining
                      ....
      In an equation for ‘chunkedTransferEncoding’:
          chunkedTransferEncoding innerBuilder
            = B.builder transferEncodingStep
            where
                transferEncodingStep :: forall a. BuildStep a -> BuildStep a
                transferEncodingStep k
                  = go (B.runBuilder innerBuilder)
                  where
                      go :: (BufferRange -> IO (BuildSignal _x)) -> BuildStep a
                      go innerStep (BufferRange op ope)
                        | outRemaining < minimalBufferSize
                        = pure $ B.bufferFull minimalBufferSize op (go innerStep)
                        | otherwise
                        = B.fillWithBuildStep innerStep doneH fullH insertChunkH brInner
                        where
                            ...
    • Relevant bindings include
        k :: BuildStep a
          (bound at Data/ByteString/Builder/HTTP/Chunked.hs:114:26)
        transferEncodingStep :: BuildStep a -> BuildStep a
          (bound at Data/ByteString/Builder/HTTP/Chunked.hs:114:5)
        innerBuilder :: Builder
          (bound at Data/ByteString/Builder/HTTP/Chunked.hs:110:25)
        chunkedTransferEncoding :: Builder -> Builder
          (bound at Data/ByteString/Builder/HTTP/Chunked.hs:110:1)
Data/ByteString/Builder/HTTP/Chunked.hs:97: failure in expression `let f = B.toLazyByteString . chunkedTransferEncoding . B.lazyByteString'
expected:
 but got:
          ^
          <interactive>:27:30: error:
              Variable not in scope:
                chunkedTransferEncoding :: Builder -> Builder

Examples: 5  Tried: 3  Errors: 0  Failures: 1

Test suite doctests: FAIL
Test suite logged to: dist/test/bsb-http-chunked-0.0.0.4-doctests.log
```

If I invoke `ghci` the same way as `tests/Doctests.hs` invokes `doctest` then I get the same error:
```bash
$ cabal exec -- ghci -isrc Data/ByteString/Builder/HTTP/Chunked.hs
Loaded package environment from /home/sol/sandbox/bsb-http-chunked/dist-newstyle/tmp/environment.-80058/.ghc.environment.x86_64-linux-9.2.1
GHCi, version 9.2.1: https://www.haskell.org/ghc/  :? for help
Loaded GHCi configuration from /home/sol/.ghci
[1 of 1] Compiling Data.ByteString.Builder.HTTP.Chunked ( Data/ByteString/Builder/HTTP/Chunked.hs, interpreted )

Data/ByteString/Builder/HTTP/Chunked.hs:117:47: error:
    • Found type wildcard ‘_x’ standing for ‘a1’
      Where: ‘a1’ is a rigid type variable bound by
               the inferred type of
                 go :: (BufferRange -> IO (BuildSignal a1)) -> BuildStep a
               at Data/ByteString/Builder/HTTP/Chunked.hs:(118,9)-(177,64)
      To use the inferred type, enable PartialTypeSignatures
    • In the type signature:
        go :: (BufferRange -> IO (BuildSignal _x)) -> BuildStep a
      In an equation for ‘transferEncodingStep’:
          transferEncodingStep k
            = go (B.runBuilder innerBuilder)
            where
                go :: (BufferRange -> IO (BuildSignal _x)) -> BuildStep a
                go innerStep (BufferRange op ope)
                  | outRemaining < minimalBufferSize
                  = pure $ B.bufferFull minimalBufferSize op (go innerStep)
                  | otherwise
                  = B.fillWithBuildStep innerStep doneH fullH insertChunkH brInner
                  where
                      outRemaining = ope `F.minusPtr` op
                      maxChunkSizeLength = word32HexLength $ fromIntegral outRemaining
                      ....
      In an equation for ‘chunkedTransferEncoding’:
          chunkedTransferEncoding innerBuilder
            = B.builder transferEncodingStep
            where
                transferEncodingStep :: forall a. BuildStep a -> BuildStep a
                transferEncodingStep k
                  = go (B.runBuilder innerBuilder)
                  where
                      go :: (BufferRange -> IO (BuildSignal _x)) -> BuildStep a
                      go innerStep (BufferRange op ope)
                        | outRemaining < minimalBufferSize
                        = pure $ B.bufferFull minimalBufferSize op (go innerStep)
                        | otherwise
                        = B.fillWithBuildStep innerStep doneH fullH insertChunkH brInner
                        where
                            ...
    • Relevant bindings include
        k :: BuildStep a
          (bound at Data/ByteString/Builder/HTTP/Chunked.hs:114:26)
        transferEncodingStep :: BuildStep a -> BuildStep a
          (bound at Data/ByteString/Builder/HTTP/Chunked.hs:114:5)
        innerBuilder :: Builder
          (bound at Data/ByteString/Builder/HTTP/Chunked.hs:110:25)
        chunkedTransferEncoding :: Builder -> Builder
          (bound at Data/ByteString/Builder/HTTP/Chunked.hs:110:1)
    |
117 |         go :: (BufferRange -> IO (BuildSignal _x)) -> BuildStep a
    |                                               ^^
Failed, no modules loaded.
```
So from what I understand, this is not an issue with `doctest`.  It's either an issue with `cabal`, `ghci` or the way `src/Doctests.hs` invokes `doctest`.

The following works and is how I recommend to use `doctest` starting from `doctest-0.20.0` with `cabal-v2`:
```bash
$ cabal install doctest --overwrite-policy=always && cabal build && cabal repl --build-depends=QuickCheck --build-depends=template-haskell --with-ghc=doctest
```

--- a/bsb-http-chunked.cabal
+++ b/bsb-http-chunked.cabal
@@ -66,6 +66,9 @@ test-suite doctests
                  , doctest >= 0.8
   ghc-options: -Wall
   type: exitcode-stdio-1.0
+  if impl(ghc >= 9.2)
+    -- https://github.com/sjakobi/bsb-http-chunked/issues/38
+    buildable: False
 
 benchmark bench
   hs-source-dirs: bench

